FLT/Deformations/Categories.lean

```diff
@@ -0,0 +1,230 @@
+import FLT.Deformations.IsProartinian
+import FLT.Deformations.IsResidueAlgebra
+import Mathlib.Topology.Algebra.Algebra.Equiv
+
+universe u
+
+open CategoryTheory Function Limits IsLocalRing
+
+namespace Deformation
+
+variable (𝓞 : Type u) [CommRing 𝓞]
+
+/-- A local proartinian algebra is a topological ring that is local and proartinian,
+and that the induced map on the residue fields is bijective. -/
+class IsLocalProartinianAlgebra
+    (R : Type u) [CommRing R] [TopologicalSpace R] [Algebra 𝓞 R] : Prop extends
+  IsTopologicalRing R, IsLocalRing R, IsProartinian R,
+  IsLocalHom (algebraMap 𝓞 R), IsResidueAlgebra 𝓞 R
```

Comment:
But not that O is local?

Reply:
We're defining a category of `R`'s, with 𝓞 fixed, so I don't think it's necessary/appropriate to include a condition solely about `R` here. Leaving out the condition makes the result more general (at least syntactically).